Automattic/wp-calypso
# A4A: Sites dashboard - Fixing Lightbox functionality for Boost within the Preview Pane

Related to https://github.com/Automattic/automattic-for-agencies-dev/issues/139

## Proposed Changes

* This PR fixes Boost functionality issues within the Sites Dashboard preview pane, in the Boost tab.
* The approach I've taken makes existing agency dashboard components A4A compatible rather than copying them over, as there are quite a few files that would need copying over and that may be better done when moving to focus exclusively on A4A.
* Specifically, the PR:
  * Hooks up the lightbox functionality (`BoostInfoModal`)
  * Fixes the Lightbox Upgrade link routes to go to the marketplace/product page with a query param including the site ID (nothing that the actual siteID implementation isn't added yet in the marketplace but the link does take you there and add the product anyway).
  * Ensures the 'blank state' for the Boost preview pane is replaced by the 'Get Score' box, which opens an upsell.
  * Additionally, fixes the 'Start for free' button which shows when clicking 'Get Score' on the sites dashboard itself. 
  

What isn't included:
* An additional issue on the sites dashboard itself relates to the pages visited when clicking on Boost score. There is an issue for that here: https://github.com/Automattic/automattic-for-agencies-dev/issues/205

## Testing Instructions

For the below testing steps, make sure to run `yarn start-a8c-for-agencies` locally and then visit `http://agencies.localhost:3000/sites`.

In addition to the below test scenarios, you can also click on 'Get score' from the sites dashboard for a site that has this showing, and in the lightbox that now opens the 'Start for free' button should now work properly (installs boost)

There are multiple testing scenarios, depending on the kind of site, the state of Boost on the site, and whether a paid plan is in place. For each of these, open the preview pane for the relevant site:

* WPcom site without Boost installed: 'Get Score' default state - links to Lightbox - the purchase link sends you to the JP dashboard first (to get Boost) - `/wp-admin/admin.php?page=jetpack#/dashboard`
* WPcom site (with Boost installed and active): Upgrade to Auto-optimize button leads to lightbox, lightbox has 'Upgrade to Auto-optimize' button, which links to the Jetpack dashboard page I mentioned above - `/wp-admin/admin.php?page=jetpack#/dashboard`
* WPcom site with paid plan: Optimize performance button leads to `wp-admin/admin.php?page=jetpack-boost`
* Pressable site with Boost installed and also without Boost installed, also one with paid plan: Upgrade to Auto-optimize button leads to lightbox with upsell (and displaying upsell price above). Clicking the button takes you to the marketplace/product page with the product and site ID in the URL, adds the item to the basket. Note that the site ID functionality isn't properly hooked up yet on the pricing page side.
* Self hosted site without Boost installed: 'Get Score' default state - links to Lightbox - the purchase link sends you to the JP dashboard first (to get Boost) - `/wp-admin/admin.php?page=jetpack#/dashboard`
* Self hosted site without Boost installed but having visited My Jetpack so initial Boost scores are calculated: Button is 'Upgrade to Auto-optimize', leads to lightbox and 'Upgrade to Auto-optimize' button, displays upsell price above, links to marketplace.
* Self hosted site with Boost and paid plan: Button is 'Optimize performance' - leads to wp-admin - `/wp-admin/admin.php?page=jetpack-boost`

Make sure as well to test from the Jetpack cloud `/dashboard` to make sure the existing functionality there is not impacted.


**Screenshots:**


Empty state: 

<img width="744" alt="Screenshot 2024-04-03 at 16 27 15" src="https://github.com/Automattic/wp-calypso/assets/16754605/3f86bd41-ac29-431c-9b4b-7d222eb2ed8b">

Auto-optimize button generally leads to:


<img width="400" alt="Screenshot 2024-04-03 at 16 34 59" src="https://github.com/Automattic/wp-calypso/assets/16754605/e6231053-92ed-4749-b9cc-f739bcb0b17c">

On a WPcom site with Boost installed and active (but no plan), Auto-optimize button leads to:

<img width="400" alt="Screenshot 2024-04-03 at 16 36 12" src="https://github.com/Automattic/wp-calypso/assets/16754605/583caf7c-e12d-4323-9b9c-94dd4926fc5f">

On WPcom site with Boost not installed yet, Auto-optimize button leads to:

<img width="400" alt="Screenshot 2024-04-03 at 16 38 17" src="https://github.com/Automattic/wp-calypso/assets/16754605/43535c5c-1c13-4fbb-a74f-b8c0f2c4ccfa">


For sites showing the Upgrade to Auto-optimize button on the sites preview pane:

<img width="400" alt="Screenshot 2024-04-03 at 16 53 23" src="https://github.com/Automattic/wp-calypso/assets/16754605/f82030c8-e916-4bf4-9413-e511a70801d3">


On sites with a paid Boost plan, the sites preview pane shows:


<img width="400" alt="Screenshot 2024-04-03 at 16 56 17" src="https://github.com/Automattic/wp-calypso/assets/16754605/ccdf197b-2217-47ab-b2c4-f2c13dafc224">



## Pre-merge Checklist
